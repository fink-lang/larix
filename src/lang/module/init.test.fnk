{describe, it, expect, to_match_snapshot} = import '@fink/jest/test.fnk'

{parse} = import '../../parser.fnk'


describe 'module', fn:

  it 'parses module', fn:
    expect
      parse '
        # leading module comment

        # expr comment
        foo = bar
        spam = ni

        # trailing module comment
      '
      to_match_snapshot
