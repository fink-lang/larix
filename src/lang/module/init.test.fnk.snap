// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`module parses module 1`] = `
module (1:0-8:0)
  # leading module comment
  # expr comment
  assign = (4:0-4:9)
    ident (4:0-4:3) foo
    ident (4:6-4:9) bar
  assign = (5:0-5:9)
    ident (5:0-5:4) spam
    ident (5:7-5:9) ni
`;
